### PR TITLE
resolve #495 docker-compose file for GeoServer

### PR DESCRIPTION
**Pull request by agpenton**
- docker-compose added with 3 services (GeoServer/ Postgres/ proxy).
- service GeoServer with `GEOSERVER_WEBAPP_SRC` build argument.
- Postgres with database initialization and healthcheck.
- nginx with the configuration for GeoServer, and port 80 mapped to the host.

### CODE REVIEW

**Comment by lpasquali** [comment]
@agpenton cloud you remove anything it is not the docker-compose and resolve conlicts?

**Comment by agpenton** [comment]
> @agpenton cloud you remove anything it is not the docker-compose and resolve conlicts?

@lpasquali resolved.

**Review by lpasquali — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - ./postgres/01-init-user.sh:/docker-entrypoint-initdb.d/01-init-user.sh
+    networks:
+      - intergration-test
+
+  geoserver:
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+      args:
+        GEOSERVER_WEBAPP_SRC: "https://build.geoserver.org/geoserver/master/geoserver-master-latest-war.zip"
```
Comment (randomorder) [inline docker-compose.yml:30]: master branch has been renamed to "main"

File: postgres/01-init-user.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -e
+
+psql -v ON_ERROR_STOP=1 --username ${POSTGRES_USER} --dbname ${POSTGRES_DB} <<-EOSQL
```
Comment (randomorder) [inline postgres/01-init-user.sh:4]: please add Postgis extension

**Review by randomorder — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (agpenton) [reply docker-compose.yml]: solved here https://github.com/geosolutions-it/docker-geoserver/pull/43/commits/329db3edae5f726910e2dc637206841b6e51af9a

File: postgres/01-init-user.sh (hunk 2 above)
Comment (agpenton) [reply postgres/01-init-user.sh:4]: solved here https://github.com/geosolutions-it/docker-geoserver/pull/43/commits/329db3edae5f726910e2dc637206841b6e51af9a

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+version: "3"
+networks:
+  intergration-test:
```
Comment (randomorder) [inline docker-compose.yml:3]: "integration-tests" ?

**Comment by randomorder** [comment]
can you please test the docker-compose @lpasquali ?

**Comment by randomorder** [comment]
@lpasquali ?

File: docker-compose.yml (hunk 3 above)
Comment (agpenton) [reply docker-compose.yml]: solved here https://github.com/geosolutions-it/docker-geoserver/pull/43/commits/dd8970b203ef4e7bee74c4db00dea8cc284f4072

**Comment by agpenton** [comment]
@randomorder the PR is updated.

**Review by lpasquali — APPROVED** [review]

**Comment by randomorder** [comment]
Should we close this one in favor of #64, @lpasquali?

**Comment by lpasquali** [comment]
> @lpasquali ?

it is merge-able, I did not do a docker-compose.yml in #64 @randomorder

**Pull request merged**